Firestore/Example/Tests/Util/FSTHelpers.mm

```diff
@@ -298,29 +299,126 @@ NSComparator FSTTestDocComparator(const absl::string_view fieldPath) {
       .snapshot;
 }
 
+@implementation FSTTestTargetMetadataProvider {
+  std::unordered_map<FSTTargetID, DocumentKeySet> _syncedKeys;
+  std::unordered_map<FSTTargetID, FSTQueryData *> _queryData;
+}
+
++ (instancetype)providerWithSingleResultForKey:(firebase::firestore::model::DocumentKey)documentKey
```

Comment:
Nit: you can use `using` declarations in the `.mm` files.

Reply:
Done.